### PR TITLE
[SettingsBundle] Recompute doctrine change-set after transforming settings

### PR DESCRIPTION
**Pull request by aramalipoor**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

When transforming our theme settings I expected to see ID to be saved but instead we saw,
```
You cannot serialize or unserialize PDO instances
```

It was because you have to tell doctrine to re-compute the entity's change-set!

### CODE REVIEW

File: src/Sylius/Bundle/SettingsBundle/Doctrine/EventListener/ParameterTransformerListener.php [hunk 1]
```diff
@@ -60,13 +61,13 @@ public function onFlush(OnFlushEventArgs $event)
...
             if ($entity instanceof SettingsInterface) {
-                $this->transform($entity);
+                $this->transform($entity, $em);
             }
         }
 
         foreach ($uow->getScheduledEntityUpdates() as $entity) {
             if ($entity instanceof SettingsInterface) {
-                $this->transform($entity);
+                $this->transform($entity, $em);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/SettingsBundle/Doctrine/EventListener/ParameterTransformerListener.php]: Please, don't abbreviate

File: src/Sylius/Bundle/SettingsBundle/Doctrine/EventListener/ParameterTransformerListener.php [hunk 2]
```diff
@@ -99,6 +101,9 @@ protected function transform(SettingsInterface $settings)
                 $settings->set($name, $transformers[$name]->transform($value));
             }
         }
+
+        $md = $em->getClassMetadata(get_class($settings));
```
Comment (lchrusciel) [inline src/Sylius/Bundle/SettingsBundle/Doctrine/EventListener/ParameterTransformerListener.php]: It is not that easy to figure out what `md` means

**Comment by michalmarcinkowski** [comment]
Thanks Aram! :+1: Restarted the PHP 5.6 build.

**Comment by Niiko** [comment]
:+1: Just ran into that issue after update :)

**Pull request merged**